amazon-qldb-driver-core/src/driver.rs

```diff
@@ -82,15 +96,12 @@ impl QldbDriverBuilder {
             .max_size(self.max_concurrent_transactions)
             .connection_timeout(Duration::from_secs(10))
             .error_sink(Box::new(QldbErrorLoggingErrorSink::new()))
-            .build(QldbSessionManager {
-                client: client.clone(),
-                ledger_name: ledger_name.clone(),
-            })
-            .await?;
+            .build(QldbSessionV2Manager::new(client, ledger_name.clone()))
+            .await
+            .map_err(|_| error::todo_stable_error_api())?;
```

Comment:
This is mapping the error variant from bb8 `Builder.build`, not an aws sdk error, right? 

And map_err results in another `Result`, rather than the success variant of the prior Result rather than being some macro which can return an error? I'm a bit lost about how we get from here to building the Ok result with a concrete session_pool.

Reply:
Yeah. So..

```rust
pub async fn build(self, manager: M) -> Result<Pool<M>, M::Error> {
```

The error variant is the associated `Error` type, i.e. this one:

```rust
#[async_trait]
impl<C> ManageConnection for QldbSessionV2Manager<C>
where
    C: SmithyConnector,
{
    type Connection = QldbHttp2Connection;
    type Error = ConnectionError;
```

and `ConnectionError` is defined by us to be:

```rust
pub type ConnectionError = SdkError<SendCommandError>;
```

So it's just the underlying SdkError that represents the issue - connection, credentials, invalid endpoint etc.

Previously, I had the carrier infrastructure (`?` operator) turn that into `QldbError::SdkError`. As per our discussion last week, you convinced me to abstract the SDK errors away. So I just put a todo there to remind myself.

`map_err` is a function that turns `Result<T, E1>` to `Result<T, E2>` (as you said). So it doesn't mess with the pool at all, just the error. The `T` itself is the pool, and the carrier trait essentially expands to:

```rust
let pool = match stuff() {
  Ok(it) => it,
  Err(err) => return Err(err).into(),
};
```